### DISCUSSION TITLE
Possibly add facilities for missing values into R-Instat?

### DISCUSSION THREAD

**Discussion by rdstern** [IDEMSInternational/R-Instat#7592] [Ideas]
We are addressing this topic (currently only very incompletely) in the climatic menu.  Here I propose we add more facilities for missing values, including infilling, into our **_prepare_** menu.

I am assuming that @lilyclements will contribute to this discussion.  It relates also to her recent PhD topic.  

a) We already have a package, written by @dannyparsons on the topic, though we don't yet use it, within R-Instat.  
b) Our options for coping with missing, extend those offered by the usual R, in an impressive way.

There is an excellent article [here](https://rpubs.com/odenipinedo/dealing-with-missing-data-in-R):  by Nicholas Tierney in 2021.  This shows that just part of the topic is the infilling.  I wonder how we could follow the sections in this article, with R-Instat.  Perhaps we should now be adding `naniar` to R-Instat?  For doing the actual imputation it uses the recent `simputation` package.

Should this be a separate section in the Prepare menu.  I assume it is mainly to deal with missing values in numeric variables, (so it could go there) but perhaps not?

**Comment by lilyclements** [comment]
I think having more on missing values in R-Instat would be great. Would you see it getting it's own "sub menu section" in the Prepare menu, or would you see it elsewhere in there, such as under Data Frame or Check Data? 
I would want to be careful on the imputation front - it feels too easy to replace missing values with the mean and not overly thinking about it (although the alternative is to just ignore them at the moment which isn't ideal). Perhaps we initially focus on checking missing values / missing values patterns (e.g MAR) with parts in the `naniar` and `visdat` packages (and I think there's another which is really good - I'll check the name of it.

**Comment by rdstern** [comment]
@lilyclements thanks for the addition.  I would like to start by being able to cope with multiple missing values in R-Instat, at least when they arrive from SPSS, etc.  @volloholic thought this was trivial, so I could handle it, but it seems not to be sufficiently trivial for me.  I am keen on this aspect, to be able to make the point in the comparison with an ordinary spreadsheet, that statistics packages cope better with missing values and you can get messed up if you just use an ordinary spreadsheet.
